File: for.py
# for - loop, structura repetitiva
# de unde incepem?
# pana unde mergem?
# pasul pe care il parcurge

# # problema: printam 101 dalmatieni din 2 in 2
# for i in range(1, 102, 2):
#     print(f'Dalmatianul cu nr {i}')
#
# # parcurgere lista cu for prin intermediul indexului
# numere = [3, 7, 10, 15, 32]
# for i in range(0, len(numere)): # i = indexul
#     print(f'indexul curent este {i} iar numarul curent este {numere[i]}')
#
# # for each - parcurge toate elementele si ajunge direct la valoare
# for numar in numere:
#     print(f'Numarul curent este: {numar}')
#
# # parcurgem array de culori
# culori = ['alb', 'rosu', 'galben']
# for culoare in culori:
#     print(f'Culoarea actuala este {culoare}')
#
# # suma (s) numerelor dintr-un array
# s = 0
# for numar in numere:
#     print(f'Numarul curent este {numar}')
#     s = s + numar
# print(f'Suma numerelor este {s}')

# de cate ori apare 3 intr-o lista de numere
lista = [3, 2, 3, 5, 3, 3, 7, 8]
numar_cautat = int(input(f"Introdu numărul pe care dorești să-l cauți în lista urmatoare {lista}: "))
numar_aparitii = 0
for numar in lista:
    if numar == numar_cautat:
        numar_aparitii += 1
print(f"Numărul {numar_cautat} apare de {numar_aparitii} ori în lista.")

File: src/For.java
public class For {
    public static void main(String[] args) {
        // for - loop, structura repetitiva
        // de unde incepem?
        // pana unde mergem?
        // pasul pe care il parcurge

        // problema: printam 101 dalmatieni
        for (int i = 1; i <= 101; i++) { //i++ = i+1
            // i = 1 apoi i = 2 pana la i = 101; la i = 102 iasa din loop pt ca 102 > 101
            System.out.println("dalmatianul cu nr " + i);
        }

        //ne ajuta sa parcurgem un array prin intermediul indexului
        int[] numere = {3, 7, 10, 15, 32}; //numere[0] = 3; avem in spate index
        for (int i = 0; i < numere.length; i = i + 2) { //parcurgem din 2 in 2
            System.out.println("elementul are index " + i + " si valoarea " + numere[i]);
        }

        //for each - parcurge toate elementele si ajunge direct la valoare
        for (int numar : numere) {
            System.out.println("Numarul este: " + numar);
        }

        //parcurgem array de culori
        String[] culori = {"alb", "rosu", "galben"};
        for (String culoare : culori) {
            System.out.println("Culoarea actuala este " + culoare);
        }

        // suma (s) numerelor dintr-un array
        int s = 0;
        for (int numar : numere) {  //s=3, 7
            s = s + numar; //0+3=3, 3+7=10, 10+10=20, 20+15=35, 35+32=67
        }
        System.out.println(s);
    }
}